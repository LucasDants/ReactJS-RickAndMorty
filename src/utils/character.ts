export const characterMock = {
    data: {
        id: '7',
        name: 'Abradolf Lincler',
        species: 'Human',
        type: 'Genetic experiment',
        gender: "Male",
        status: 'unknown',
        episode: [{
            "id": "10"
        },
        {
            "id": "11"
        }
        ],
        image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        origin: {
            name: 'Earth (Replacement Dimension)'
        }
    }
}