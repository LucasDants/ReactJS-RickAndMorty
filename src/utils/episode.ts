export const episodeMock = {
    data: {
        id: '1',
        name: 'Pilot',
        air_date: 'December 2, 2013',
        episode: 'SO1E01',
        characters: [
            {
                "name": "Rick Sanchez"
            },
            {
                "name": "Morty Smith"
            },
            {
                "name": "Bepisian"
            },
            {
                "name": "Beth Smith"
            },
            {
                "name": "Canklanker Thom"
            },
            {
                "name": "Davin"
            },
            {
                "name": "Frank Palicky"
            },
            {
                "name": "Glenn"
            },
            {
                "name": "Hookah Alien"
            },
            {
                "name": "Jerry Smith"
            },
            {
                "name": "Jessica"
            },
            {
                "name": "Jessica's Friend"
            },
            {
                "name": "Mr. Goldenfold"
            },
            {
                "name": "Mrs. Sanchez"
            },
            {
                "name": "Principal Vagina"
            },
            {
                "name": "Summer Smith"
            },
            {
                "name": "Davin"
            },
            {
                "name": "Greebybobe"
            },
            {
                "name": "Pripudlian"
            }
        ]
    }
}