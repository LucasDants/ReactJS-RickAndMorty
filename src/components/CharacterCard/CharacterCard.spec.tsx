import { render } from "@testing-library/react"
import { CharacterCard } from "."
import { characterMock } from "../../utils/character"

describe('CharacterCard Component', () => {
    it('should renders correctly', () => {
        const view = render(
            <CharacterCard data={characterMock.data} />
        )
        expect(view).toMatchSnapshot()
    }) 
})