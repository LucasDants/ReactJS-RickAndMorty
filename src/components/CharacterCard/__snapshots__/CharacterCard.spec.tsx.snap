// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard Component should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1bsv1wv"
      >
        <img
          class="chakra-image css-x8fuok"
          src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        />
        <div
          class="css-1q3bl5i"
        >
          <p
            class="chakra-text css-18fe8az"
          >
            Name
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              Abradolf Lincler
            </span>
          </p>
          <p
            class="chakra-text css-18fe8az"
          >
            Species
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              Human
            </span>
          </p>
          <p
            class="chakra-text css-18fe8az"
          >
            Type
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              Genetic experiment
            </span>
          </p>
          <p
            class="chakra-text css-18fe8az"
          >
            Gender
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              Male
            </span>
          </p>
          <p
            class="chakra-text css-18fe8az"
          >
            Status
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              unknown
            </span>
          </p>
          <p
            class="chakra-text css-18fe8az"
          >
            Origin
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              Earth (Replacement Dimension)
            </span>
          </p>
          <p
            class="chakra-text css-18fe8az"
          >
            Episodes
            : 
            <span
              class="chakra-text css-1au9iyi"
            >
              10, 11
            </span>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1bsv1wv"
    >
      <img
        class="chakra-image css-x8fuok"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <div
        class="css-1q3bl5i"
      >
        <p
          class="chakra-text css-18fe8az"
        >
          Name
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            Abradolf Lincler
          </span>
        </p>
        <p
          class="chakra-text css-18fe8az"
        >
          Species
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            Human
          </span>
        </p>
        <p
          class="chakra-text css-18fe8az"
        >
          Type
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            Genetic experiment
          </span>
        </p>
        <p
          class="chakra-text css-18fe8az"
        >
          Gender
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            Male
          </span>
        </p>
        <p
          class="chakra-text css-18fe8az"
        >
          Status
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            unknown
          </span>
        </p>
        <p
          class="chakra-text css-18fe8az"
        >
          Origin
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            Earth (Replacement Dimension)
          </span>
        </p>
        <p
          class="chakra-text css-18fe8az"
        >
          Episodes
          : 
          <span
            class="chakra-text css-1au9iyi"
          >
            10, 11
          </span>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
